### PR TITLE
Implement a wavelength correction based on twilight frames

### PR DESCRIPTION
**Pull request by PabloCorcho**
I have created a first draft of the wavelength correction based on twilights. It works with both blue and red arms (almost automatically), although I need to improve the performance a little bit.

### CODE REVIEW

File: src/pykoala/corrections/sky.py [hunk 1]
```diff
@@ -241,6 +243,12 @@ def medfilt_continuum(data, window_size=5):
         continuum = scipy.signal.medfilt(data, window_size)
         return continuum
 
+    @staticmethod
+    def percentile_continuum(data, percentile, window_size):
+        continuum = scipy.ndimage.percentile_filter(data, percentile,
+                                                    window_size)
+        return continuum
```
Comment (paranoya) [inline src/pykoala/corrections/sky.py:250]: Why this alias?

File: test/fit_ppxf_solar_spec.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+rss = koala_rss("/home/pcorchoc/Develop/pykoala-tutorials/tutorials/data/koala/385R/27feb20012red.fits")
+fibre = 100
+# rss = koala_rss("/home/pcorchoc/Develop/pykoala-tutorials/tutorials/data/koala/580V/27feb10012red.fits")
+
+# sky_spectra /= (percentile_filter(sky_spectra, 90, 100) + 1e-100)
+velscale = np.min(constants.c.to_value("km/s")*np.diff(np.log(rss.wavelength)))  # Preserve smallest velocity step
+
+sun = fits.open("/home/pcorchoc/Research/obs_data/std_stars/calspec/sun_reference_stis_002.fits")
+sun = fits.open("/home/pcorchoc/Research/obs_data/std_stars/calspec/sun_mod_001.fits")
```
Comment (paranoya) [inline test/fit_ppxf_solar_spec.py:39]: Ouch! ;^D

File: test/fit_ppxf_solar_spec.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+
+from astropy.io import fits
+from astropy.wcs import WCS
+import numpy as np
+import matplotlib.pyplot as plt
+
+from ppxf.ppxf import ppxf
+import ppxf.ppxf_util as util
+import ppxf.sps_util as lib
```
Comment (paranoya) [inline test/fit_ppxf_solar_spec.py:17]: Do you use `util` and `lib`?

File: src/pykoala/corrections/wavelength.py [hunk 4]
```diff
@@ -102,6 +107,240 @@ def apply(self, rss):
...
+        # solar_continuum = ContinuumEstimator.medfilt_continuum(solar_spectra,
+        #                                                        delta_pixel)
+        solar_continuum = median_filter(solar_spectra, delta_pixel)
+
+        # Detect absorption features
+        median_continuum_ratio = np.nanmedian(solar_spectra / solar_continuum)
+        weights = np.abs(solar_spectra / solar_continuum -  median_continuum_ratio)
+        return weights
+
+    def compute_grid_of_models(self, pix_shift_array, pix_std_array, pix_array,
```
Comment (paranoya) [inline src/pykoala/corrections/wavelength.py:146]: Perhaps not the most efficient way, but we can worry about that later

File: src/pykoala/corrections/wavelength.py [hunk 5]
```diff
@@ -102,6 +107,240 @@ def apply(self, rss):
...
+                ) for fibre in data_container.rss_intensity])
+        else:
+            new_wavelength = data_container.wavelength
+            rss_intensity = data_container.rss_intensity
+        
+        # Interpolate the solar spectrum to the new grid of wavelengths
+        sun_intensity = flux_conserving_interpolation(
+        new_wavelength, self.sun_wavelength, self.sun_intensity)
+
+        # Make an array of weights to focus on the absorption lines
```
Comment (paranoya) [inline src/pykoala/corrections/wavelength.py:193]: What about sky emission lines?

File: src/pykoala/corrections/sky.py (hunk 1 above)
Comment (PabloCorcho) [reply src/pykoala/corrections/sky.py:250]: Why not?

File: src/pykoala/corrections/wavelength.py (hunk 5 above)
Comment (PabloCorcho) [reply src/pykoala/corrections/wavelength.py:220]: There are no sky lines in the twilight spectra, the problematic ones are the tellurics

File: src/pykoala/corrections/sky.py (hunk 1 above)
Comment (PabloCorcho) [reply src/pykoala/corrections/sky.py:250]: Let's leave as it is now, and we can revisit this later on.

**Pull request merged**